language-support/ts/codegen/src/Main.hs

```diff
@@ -92,8 +92,14 @@ genModule mod
             ["// Generated from " <> T.intercalate "/" (unModuleName curModName) <> ".daml"
             ,"/* eslint-disable @typescript-eslint/camelcase */"
             ,"/* eslint-disable @typescript-eslint/no-use-before-define */"
-            ,"import * as daml from '@digitalasset/daml-json-types';"
+            ,"/* eslint-disable @typescript-eslint/no-namespace */"
+            -- We use namespaces to add decoders to enumerations. eslint prefers modules, but
+            -- these would have to live in "d.ts" files and are restricted in what they can contain.
             ,"import * as jtv from '@mojotech/json-type-validation';"
+            ] ++
+            [
+            "import * as daml from '@digitalasset/daml-json-types';"
+            | moduleNameString curModName `notElem` ["GHC.Types", "DA.Date"]
```

Comment:
Why do we need this? I'd be very much in favor of dropping this exception.

Reply:
I think I know what the problem is. This could hit us on other modules too. Let's definitely drop this and instead add a way to assert that the generated object for the enum type implements the `Serializable` interface. This will make the "unused imports" warning go away and give us more certainty that we don't accidentally break the codegen.

Reply:
Unfortunately, we can't declare that a namespace implements an interface. The best I could come up with to work around this fact is as follows:
1. Add a function `const STATIC_IMPLEMENTS_SERIALIZABLE_CHECK = <T>(_: Serializable<T>) => { }` to `daml-json-types`.
2. After the declaration of each enum, say `Color`, add a line `STATIC_IMPLEMENTS_SERIALIZABLE_CHECK<Color>(Color);`. This line kills two birds with one stone: It ensures that `Color` actually implements `Serialiazble<Color>` and it uses the import `daml-json-types`.

Reply:
good idea, I'll try it!